docs/DeveloperGuide.md

```diff
@@ -1,15 +1,247 @@
+
+**3. UI**
+
+UI class is in charge of the majority of the print functions present in the program.
+It is instantiated once in both Parser and Duke classes, where it's print functions are utilized
+to print outputs to the User.
+
+Future Improvements: UI to handle ALL of the printing functions present in the program.
+
+**4. Storage**
+
+Storage class holds an ArrayList of modules that the user has selected.
+It saves the user's modules to an external .txt file every time the module list is altered,
+and reads from the same .txt file when the program is first booted up.
+
+**5. DataReader**
+
+DataReader class reads two external .txt files to acquire the list of Partner Universities and list
+of Modules available in the PUs, and provides this information to other components.
+
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Storage
+
+The Storage class handles the loading and storing of information into a single text file in the User's computer.
+During the initialisation of the Storage class, which is at the start of the program, loading of data occurs.
+From here on, querying of the Storage class is allowed, where modules saved by the user can be accessed through
+the ListCurrentCommand. There are only two commands that will cause changes to the text file. They are the AddCommand
+and DeleteCommand. This involves adding of new modules and deleting of old modules to the Storage. The text file will
+be continuously updated every time an Add or Delete command is called.
+
+How module data is stored in text file:
+
+Module information is stored in one single line separated by commas
+`univID`,`moduleCode`,`moduleName`,`moduleMCs`,`nusModuleCode`,`nusModuleName`,`nusModuleMcs`
+
+Sequence Diagram of Storage initialisation:
+
+![Storage.png](diagrams%2FStorage%2FStorage.png)
+
+The Storage class also handles the adding of new modules and the deleting of past modules. When any of this occurs, the
+txt file will be updated immediately after the successful adding/deletion of saved modules in the Storage.
+
+- For adding of newly saved modules, they are added through appending to the saved_modules.txt file
+- For deleting of past saved modules, they are deleted, and the txt file is updated by rewriting every module from the
+  ArrayList of saved modules
+
+### Parser
+
+The parser class is responsible for parsing the user's input commands and returning the appropriate command object.
+The commands that the parser class will initialise are ListPuCommand(), ListCurrentCommand(modules),
+prepareListPuModulesCommand(userCommandSecondKeyword, universities), ExitCommand(),
+prepareAddModuleCommand(storage, userCommandSecondKeyword, puModules, universities),
+DeleteModuleCommand(storage, indexToRemove, modules), and HelpCommand(). The parser class will handle error checking by
```

Comment:
Perhaps you can consider reformatting this to make it easier to read, as all the commands are lumped together and messy.